packages/core/src/services/scales-cartesian.ts

```diff
@@ -53,14 +53,44 @@ export class CartesianScales extends Service {
 
 	protected domainAxisPosition: AxisPositions;
 	protected rangeAxisPosition: AxisPositions;
+	protected secondaryDomainAxisPosition: AxisPositions;
+	protected secondaryRangeAxisPosition: AxisPositions;
+
+	protected dualAxes: Boolean;
 
 	protected orientation: CartesianOrientations;
 
-	getDomainAxisPosition() {
+	getDomainAxisPosition(datum?: any) {
+		if (this.dualAxes && datum) {
+			const options = this.model.getOptions();
+			const { groupMapsTo } = options.data;
+			const axesOptions = Tools.getProperty(options, "axes", this.secondaryDomainAxisPosition);
+			const dataset = datum[groupMapsTo];
+			if (axesOptions?.correspondingDatasets && axesOptions.correspondingDatasets.includes(dataset)) {
+				return this.secondaryDomainAxisPosition;
+			}
+		}
 		return this.domainAxisPosition;
 	}
 
-	getRangeAxisPosition() {
+	getRangeAxisPosition({datum = null, groups = null} = {}) {
+		if (this.dualAxes && datum) {
```

Comment:
if `datum` is falsey, then your `else if (groups && groups.length > 0)` wouldn't run bc `else if (groups && groups.length > 0)` wouldn't be satisfied